src/field/pokemon.ts

```diff
@@ -1038,12 +1038,10 @@ export abstract class Pokemon extends Phaser.GameObjects.Container {
    * @param value the desired numeric value
    */
   setStatStage(stat: BattleStat, value: number): void {
-    if (this.summonData) {
-      if (value >= -6) {
-        this.summonData.statStages[stat - 1] = Math.min(value, 6);
-      } else {
-        this.summonData.statStages[stat - 1] = Math.max(value, -6);
-      }
+    if (value >= -6) {
+      this.summonData.statStages[stat - 1] = Math.min(value, 6);
+    } else {
+      this.summonData.statStages[stat - 1] = Math.max(value, -6);
     }
```

Comment:
I just noticed this can be solved with `Phaser.Math.Clamp` which makes the `if` obsolete:

```suggestion
   this.summonData.statStages[stat - 1] = Phaser.Math.Clamp(value, -6, 6);
```

https://docs.phaser.io/phaser/concepts/math#clamp